tools/packaging/kata-deploy/local-build/cache-kata-deploy-binaries.sh

```diff
@@ -0,0 +1,285 @@
+	local guest_image_last_commit="$(get_last_modification "${repo_root_dir}/tools/packaging/guest-image")"
+	local agent_last_commit="$(get_last_modification "${repo_root_dir}/src/agent")"
+	local libs_last_commit="$(get_last_modification "${repo_root_dir}/src/libs")"
+	local gperf_version="$(get_from_kata_deps "externals.gperf.version")"
+	local libseccomp_version="$(get_from_kata_deps "externals.libseccomp.version")"
+	local rust_version="$(get_from_kata_deps "languages.rust.meta.newest-version")"
+	local rootfs_tarball_name="kata-static-rootfs-${ROOTFS_FLAVOUR}.tar.xz"
+	local current_rootfs_version="${osbuilder_last_commit}-${guest_image_last_commit}-${agent_last_commit}-${libs_last_commit}-${gperf_version}-${libseccomp_version}-${rust_version}-${ROOTFS_FLAVOUR}"
+	create_cache_asset "${rootfs_tarball_name}" "${current_rootfs_version}" ""
+}
+
+cache_shim_v2_artifacts() {
+	local shim_v2_tarball_name="kata-static-shim-v2.tar.xz"
+	local shim_v2_last_commit="$(get_last_modification "${repo_root_dir}/src/runtime")"
+	local protocols_last_commit="$(get_last_modification "${repo_root_dir}/src/libs/protocols")"
+	local runtime_rs_last_commit="$(get_last_modification "${repo_root_dir}/src/runtime-rs")"
+	local golang_version="$(get_from_kata_deps "languages.golang.meta.newest-version")"
+	local rust_version="$(get_from_kata_deps "languages.rust.meta.newest-version")"
+	local current_shim_v2_version="${shim_v2_last_commit}-${protocols_last_commit}-${runtime_rs_last_commit}-${golang_version}-${rust_version}"
+	local current_shim_v2_image="$(get_shim_v2_image_name)"
+	create_cache_asset "${shim_v2_tarball_name}" "${current_shim_v2_version}" "${current_shim_v2_image}"
+}
+
+cache_virtiofsd_artifacts() {
+	local virtiofsd_tarball_name="kata-static-virtiofsd.tar.xz"
+	local current_virtiofsd_version="$(get_from_kata_deps "externals.virtiofsd.version")-$(get_from_kata_deps "externals.virtiofsd.toolchain")"
+	local current_virtiofsd_image="$(get_virtiofsd_image_name)"
+	create_cache_asset "${virtiofsd_tarball_name}" "${current_virtiofsd_version}" "${current_virtiofsd_image}"
+}
+
+create_cache_asset() {
+	local component_name="${1}"
+	local component_version="${2}"
+	local component_image="${3}"
+
+	local build_dir="${repo_root_dir}/tools/packaging/kata-deploy/local-build/build"
+	local cache_dir="${script_dir}/cache"
+
+	sudo chown -R "${USER}:${USER}" "${build_dir}/${component_name}"
+
+	mkdir -p ${cache_dir}/${cache_target}
+	cp "${build_dir}/${component_name}" "${cache_dir}/${cache_target}/${component_name}"
+	sha256sum "${build_dir}/${component_name}" > "${cache_dir}/${cache_target}/sha256sum"
+	cat "${cache_dir}/${cache_target}/sha256sum"
+	echo "${component_version}" > "${cache_dir}/${cache_target}/latest"
+	cat "${cache_dir}/${cache_target}/latest"
+	echo "${component_image}" > "${cache_dir}/${cache_target}/latest_image"
+	cat "${cache_dir}/${cache_target}/latest_image"
+
+	docker build \
```

Comment:
Docker build uses cached layers too, so I am wondering if we shouldn't pass `--no-cache` to avoid silent problems.

For example, if I build `cache_target=kernel` for two different branches, BRANCH=`main` first then BRANCH=`ccv0`,  it will reuse the built layers to build the ccv0 image, won't it?  Because the `COMPONENT` is the same then no cache miss occurs, right?